### ISSUE TITLE
No sound playing

### ISSUE THREAD

**Issue by jayden-jw** [JakeS0ft/XSTREAMOS#2]
Installed the XStreamPixel version firmware but no sound playing at all... can see the output with serial monitor everything seems ok and responded to the Main Switch and Aux switch press but just no sound playing from the speaker.

To further check if any problem on my hardware so i flashed your another project 'nRF52_SimpleWavPlayer' and surprisingly it managed to play the wav file from the SDcard without problem.

**Comment by JakeS0ft** [comment]
Did you confirm the file paths on the SD card are as expected? It's really picky. For example "font1" is OK, but "Font1" is not nor is "font01", etc.

**Comment by jayden-jw** [comment]
I follow exact what stated in the manual., it's font1. This is the output from the serial monitor, it seems no problem with the path, all wav were detected.  That's no sound at all even the menu sound is not playing.

SD init completed.
Internal file system init completed.
Settings loaded.
I2S Wav player started.
Detected 1 sound fonts.
Saber sound init complete.
Buttons ready.
Blade init complete.
Motion engine init complete.
Accent LED ready.
SM Boot.
font1/blst01.wav exists.
font1/boot01.wav exists.
font1/hum01.wav exists.
font1/lock01.wav exists.
font1/in01.wav exists.
font1/out01.wav exists.
font1/clsh01.wav exists.
font1/swng01.wav exists.
font1/swingl01.wav exists.
font1/swingh01.wav exists.
font1/font01.wav exists.
SetFont() Prefetching font1/clsh01.wav
Omni swing detected.
Trying to play font1/font01.wav
Off

**Comment by jayden-jw** [comment]
Withtout changing the hardware, i just change the pin in nRF52_SimpleWavPlaye to match the  XStream pin and i was able to play the wav file in sd card.

#include "Arduino.h"
#include "nRF52Audio.h"

#define PIN_I2S_MCK 13
#define PIN_I2S_BCLK (A4) // A4
#define PIN_I2S_LRCK (A5) // A5
#define PIN_I2S_DIN 27 // A6 18
#define PIN_I2S_SD  7 // 27 10
#define PIN_SPI_CS  11
#define PIN_SCL 16
#define PIN_SDA 15

**Comment by JakeS0ft** [comment]
I'm looking into this. Straight from the firmware code:
#define PIN_I2S_MCK 13
#define PIN_I2S_BCLK (A4)
#define PIN_I2S_LRCK (A5)
#define PIN_I2S_DIN 27
#define PIN_I2S_SD  A7
#define PIN_SPI_CS  11


The only difference I can see from your example pinout is that PIN_I2S_SD is A7 and not 7. (Pin A7 maps to pin 31 in the core we are using.) Although, I'm not sure if this matters since all the wiring diagrams show the SD pin on the I2S breakout as not wired,  but it is a difference. Did you wire the SD pin on your I2S amp to anything?

**Comment by jayden-jw** [comment]
I just test again with SD Pin connect to pin31 but still the same no sound. In nRF52_SimpleWavPlayer it works with or without SD pin being connected.

**Comment by JakeS0ft** [comment]
What if you jumper your SD pin high (3.3V)? I'm not sure what breakout you are using, so check your documentation, but I'm wondering if it's in shutdown mode because it's not being pulled high by the MCU pin.

**Comment by JakeS0ft** [comment]
We never actually had to wire the SD pin on the breakouts we used; we always just left it floating.

**Comment by jayden-jw** [comment]
> We never actually had to wire the SD pin on the breakouts we used; we always just left it floating.

Yes I followed the given diagram, just for my own test i wire the SD  but it work with or without SD wire beign wired. The thing is without changing anything on hardware it works in nRF52_SimpleWavPlayer but the same hardware configuration dens't work on xstream firmware.

**Comment by JakeS0ft** [comment]
I created new branch to try to solve this. 
https://github.com/JakeS0ft/XSTREAMOS/tree/sound-fix

I updated the home-brew pixel firmware to use Pin 7 for I2S SD just to rule out pin problems causing your issue. You can DL that and see if it makes any difference.

**Comment by jayden-jw** [comment]
The new firmware still the same, no sound. I have recorded a short clip please have a look the initial part i flashed with nRF52_SimpleWavPlayer to proof all hardware and wiring connected correctly and working with sound then i flashed the new XstreamOS firmware and no sound at all no matter which button i pressed.

https://youtu.be/dVhoyHi662o

**Comment by jayden-jw** [comment]
Wondering if this is a close source project? If not maybe you can release the source code so that I can help to troubleshoot the problem as well but if this is a close source project just ignore this.

**Comment by Protonerd** [comment]
Do you see any other response from the system? Did you already wire up the LEDs/pixel stripes and if yes, do they light up?

**Comment by Protonerd** [comment]
Moreover, did you copy the menu sound files to your SD card?

**Comment by jayden-jw** [comment]
> Moreover, did you copy the menu sound files to your SD card?

I haven't wire the pixel stripe yet, I can wire it up later to test, and yes I did copied the menu sound files and all in the menu folder, also created a font1 folder with a set of required wav file in the SD card.

**Comment by JakeS0ft** [comment]
A few more things to try:
1) Run your font through the font converter tool if you haven't done so already.
2) Connect to the GUI and check that the volume is not set to zero on the options screen.
3) Connect to the GUI and reload default settings.
4) Ensure "Homebrew Pixel" is the firmware selected for upload ("Infinity" variants are for the all-in-one boards).
5) Upload firmware with the installer script (distro/InstallFirmware.bat) instead of the GUI so you can monitor the process.
  5a) NOTE: You need to modify the script to use the COM port and firmware variant you want to use.

**Comment by jayden-jw** [comment]
Run your font through the font converter tool if you haven't done so already.
Done - Converted from the FX-saberOS sound font

Connect to the GUI and check that the volume is not set to zero on the options screen.
Done- Already set to 100 and saved.

Connect to the GUI and reload default settings.
Done- The first thing i did when i first connected.

Ensure "Homebrew Pixel" is the firmware selected for upload ("Infinity" variants are for the all-in-one boards).
Done- As shown in my earlier post in the video clip. https://youtu.be/dVhoyHi662o

Upload firmware with the installer script (distro/InstallFirmware.bat) instead of the GUI so you can monitor the process.
Done - As shown in my earlier post in the video clip. https://youtu.be/dVhoyHi662o

**Comment by JakeS0ft** [comment]
Ah, thanks for reminding me of your video. Upon closer inspection, I think I see that you are uploading at 1:02 firmware "XStreamPixel.zip". You want instead for that to be "XStreamBFPixel.zip". The "BF" version is intended for the BlueFruit boards. You can edit the batch file and upload again with this other version.

**Comment by jayden-jw** [comment]
Thanks for the tip i didnt notice the BF version is actually stand for BlueFruit even though it's quite obvious, lol. Anyway after i flased the BF version it seems to work Intermittently, i can hear the sound when i power up and the menu sound is working too but after changing a few settings in the menu the sound totally gone, i have to disconnect the power and reboot again then the sound will work again for a short while. I'm still trying to figure out what's the cause. Will update here again.

**Comment by JakeS0ft** [comment]
If you are powering by USB, you may need to turn the volume down or sound will cut out.

**Comment by jayden-jw** [comment]
> If you are powering by USB, you may need to turn the volume down or sound will cut out.

Yes, i noticed that too, I'm now running on battery to test, so far seems to be ok but speaker still having some cracking sound even with a fully charged battery. Let me try a few more different sound font.

**Comment by jayden-jw** [comment]
the sound output quality is quite bad, I'm using existing sound font from my another saber that playing quite well on the proffie board then i run through all the wav files with the InfinitySoundConverter but seems no iprovement. Try change to the speaker that also working reasonably well from another saber but same result keep having the cracking sound. I also noticed a fully charged 4.2v battery seems to be improved but as soon as the battery dropped to 4v the sound quality start getting bad.

**Comment by jayden-jw** [comment]
After few days of testing I keep getting Intermittent sound cut out even with fully charged battery, tested with a few different speaker 8ohm 2w , 4ohm 2w and some others spec result still the same especially volume above 5 and if using volume below 5 is too soft nearly impposible to hear. Other than the sound issue the rest seems working fine.

**Comment by Protonerd** [comment]
To improve the sound playback quality I can give you suggestions only, since I do not face the same problem and replicating them would be quite difficult without your setup. First off check your wires. Especially the SD card connection is crucial, the supply of the SD card can cause sound cut-offs if the core cannot access the SD card any more or not with the bandwith needed for proper data transfer. I assume that is what you experience. Proper wiring of the I2S is just as important. Even with off-the shelf simple 2W speakers the I2S delivers a loud sound. Infinity boosts it with a 5V supply, but even supplying the amp from the 3.7V battery shall be sufficient to make your neighbors jump from bed if you switch on your saber at full volume.
Regarding the sound quality, please check that the sound fonts you use are free of metadata. They can cause pops and noise in the playback if they are interpreted as sound data, which they are not. Here I just assume the bad quality you mentioned are due to pops. If not, maybe you can describe them more in detail or make a short video.

**Comment by JakeS0ft** [comment]
Another thing to note is that depending on the age and quality of your breadboard, the I2S amp might not be getting enough current even if you wired it correctly.

**Comment by jayden-jw** [comment]
![20210525_092511](https://user-images.githubusercontent.com/76594891/119426165-4dbec480-bd3b-11eb-83f4-9f3ef95bf786.jpg)
![20210525_092514](https://user-images.githubusercontent.com/76594891/119426187-57482c80-bd3b-11eb-92ac-85af1f81b358.jpg)

Just got my chassis printed ... soon I will put it all together and wire it up. Will update again if the sound and volume improve.

**Comment by JakeS0ft** [comment]
That looks real nice!

**Comment by jayden-jw** [comment]
Thanks for all the helps and suggestions,  after I wired up everything it seems working fine and no more sound cut out. Volume is a bit on the low side but I just amplify the wav loudness slightly and it's quite loud now.

**Comment by Protonerd** [comment]
Glad it works now! I guess I forgot to mention in the tutorial, that due to the currents needed the system needs hard wired supply lines to work properly. Wrt volume, it strongly depends on your setup, speaker, place in your hilt etc.

Your chassis is indeed remarkable, never seen one which incorporates both a battery holder and the switches. You made me think to experiment with similar ones myself, thanks for sharing!
